packages/@aws-cdk/aws-appsync/lib/data-source.ts

```diff
@@ -248,4 +250,37 @@ export class LambdaDataSource extends BackedDataSource {
     });
     props.lambdaFunction.grantInvoke(this);
   }
+}
+
+/**
+ * Properties for an AppSync RDS datasource
+ */
+export interface RdsDataSourceProps extends BackedDataSourceProps {
+  /**
+   * The database cluster to call to interact with this data source
+   */
+  readonly databaseCluster: IDatabaseCluster;
+  /**
+   * The secret containing the credentials for the database
+   */
+  readonly secretStore: ISecret;
+}
+
+/**
+ * An AppSync datasource backed by RDS
+ */
+export class RdsDataSource extends BackedDataSource {
+  constructor(scope: Construct, id: string, props: RdsDataSourceProps) {
+    super(scope, id, props, {
+      type: 'RELATIONAL_DATABASE',
+      relationalDatabaseConfig: {
+        rdsHttpEndpointConfig: {
+          awsRegion: props.databaseCluster.stack.region,
+          dbClusterIdentifier: props.databaseCluster.clusterIdentifier,
+          awsSecretStoreArn: props.secretStore.secretArn,
+        },
+        relationalDatabaseSourceType: 'RDS_HTTP_ENDPOINT',
+      },
```

Comment:
I believe you need a grant statement here to give `appsync` the ability to get/update/etc. the RDS cluster and to get the secret value for the secret. See the policy statement for appsync rds [here](https://docs.aws.amazon.com/appsync/latest/devguide/tutorial-rds-resolvers.html#graphql-schema).

See the dynamodb data source implementation for reference:
https://github.com/aws/aws-cdk/blob/975efbed5518c4b4b982038ee6a4ad79fe11935d/packages/%40aws-cdk/aws-appsync/lib/data-source.ts#L196-L200

----

In this case, I believe the grants needed are as follows.

For secret:
```ts
props.secretStore.grantRead(this);
```

For rds:

There is no rds grant methods at the moment. Which means we would need a PR to add a grant method in rds to provide the equivalent of:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "rds-data:DeleteItems",
                "rds-data:ExecuteSql",
                "rds-data:ExecuteStatement",
                "rds-data:GetItems",
                "rds-data:InsertItems",
                "rds-data:UpdateItems"
            ],
            "Resource": [
                "arn:aws:rds:us-east-1:123456789012:cluster:mydbcluster",
                "arn:aws:rds:us-east-1:123456789012:cluster:mydbcluster:*"
            ]
        }
    ]
}
```

@skinny85 wdyt? I think its either adding a grant or to just make a Iam policy here which im not a fan of

Reply:
spoke to @skinny85 offline and came to the conclusion the best way to do this would be to just make the policy here and put a note to change it once rds implements grants.

the implementation would look something like this
```ts
const clusterArn = cdk.Stack.of(this).formatArn({
    service: 'rds',
    resource: `cluster:${props.databaseCluster.clusterIdentifier}`,
});

iam.Grant.addToPrincipal({
  grantee: this,
  actions: [ 
     "rds-data:DeleteItems",
     "rds-data:ExecuteSql",
     "rds-data:ExecuteStatement",
     "rds-data:GetItems",
     "rds-data:InsertItems",
      "rds-data:UpdateItems",
    ],
  resourceArns: [ clusterArn,  `${clusterArn}:*` ],
  scope: this,
});

```